### PR TITLE
Adding new fast forward action

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

This adds a new custom Play Action that will be used for fast forwarding an application to the latest program version.

Right now this will not run. No controllers are configured to call it.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Created unit and/or browser tests which fail without the change (if possible)

### Issue(s) this completes

Related to: #5541

### CODE REVIEW

File: server/app/actions/ProgramFastForwardApplicationAction.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    String routePattern = request.attrs().get(Router.Attrs.HANDLER_DEF).path();
+    RouteExtractor routeExtractor = new RouteExtractor(routePattern, request.path());
+
+    long programId = routeExtractor.getParamLongValue("programId");
+    long applicantId = getApplicantId(request, routeExtractor);
+
+    long latestProgramId = programRepository.getMostRecentActiveProgramId(programId);
+
+    if (latestProgramId > programId) {
+      applicationRepository.updateDraftApplicationProgram(applicantId, latestProgramId);
```
Comment (gwendolyngoetz) [inline server/app/actions/ProgramFastForwardApplicationAction.java:48]: I am aware this does not do the `checkApplicantAuthorization` or `checkProgramAuthorization` checks like in the controllers.

That will be addressed in a separate PR and in a separate reusable action.

File: server/test/actions/ProgramFastForwardApplicationActionTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    when(profileUtilsMock.getApplicantId(request)).thenReturn(Optional.of(applicantId));
+    when(programRepositoryMock.getMostRecentActiveProgramId(any(Long.class)))
+        .thenReturn(currentProgramId);
+
+    // The action to test
+    ProgramFastForwardApplicationAction action =
+        new ProgramFastForwardApplicationAction(
+            profileUtilsMock, programRepositoryMock, applicationRepositoryMock);
+
+    // Configure a fake secondary action to serve as a stand in for the the last action in the chain
```
Comment (rockycodes) [inline server/test/actions/ProgramFastForwardApplicationActionTest.java:57]:
```suggestion
    // Configure a fake secondary action to serve as a stand in for the last action in the chain
```
```suggestion
    // Configure a fake secondary action to serve as a stand in for the the last action in the chain
```

**Review by rockycodes — APPROVED** [review]
lgtm!
